src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -31,14 +41,38 @@ impl ScalarFunc {
         let res = t.date_format(format_mask_str)?;
         Ok(Some(Cow::Owned(res.into_bytes())))
     }
+
+    #[inline]
+    pub fn date<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Time>>> {
+        let t = try_opt!(self.children[0].eval_time(ctx, row));
```

Comment:
It seems we need to handle the error as invalid time error when eval_time failed according to tidb https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L272

Reply:
Done